src/riak_ql_ddl.erl

```diff
@@ -315,9 +320,9 @@ is_filters_field_valid(Mod, {Op, Field, {RHS_type, RHS_Val}}, Acc1) ->
     case Mod:is_field_valid([Field]) of
         true  ->
             ExpectedType = Mod:get_field_type([Field]),
-            case is_compatible_type(ExpectedType, RHS_type, normalise(RHS_Val)) of
-                true  -> Acc2 = Acc1;
-                false -> Acc2 = [{incompatible_type, Field, ExpectedType, RHS_type} | Acc1]
+            Acc2 = case is_compatible_type(ExpectedType, RHS_type, normalise(RHS_Val)) of
```

Comment:
It's always better to not let variables be bound within case branches. Well spotted.